### ISSUE TITLE
Open Graph Facebook meta tags - Server error

### ISSUE THREAD

**Issue by atmar** [nuxt/nuxt.js#2034]
Hello,

When I'm testing out the open graph on our new site, it just gives us a generic nuxt server error with no extra information. You can view it at https://developers.facebook.com/tools/debug/sharing/?q=https%3A%2F%2Fstore.gameodds.gg%2Fgame%2Ffantasy-bundle . When I visit the site, all meta tags are visible and correct.

head code:
```
  head() {
    let product = this.product
    return {
      title: `${product.name} (PC) - Gameodds.gg`,
      meta: [
        {
          hid: `description`,
          name: 'description',
          content: product.short_description
        },
        {
          hid: `keywords`,
          name: 'keywords',
          keywords: product.name
        },
        {
          hid: `og:title`,
          property: 'og:title',
          content: `${product.name} (PC) - Gameodds.gg`
        },
        {
          hid: `og:url`,
          property: 'og:url',
          content: 'https://store.gameodds.gg' + this.$route.fullPath
        },
        {
          hid: `og:image`,
          property: 'og:image',
          content: `${this.$store.state.imageUrl}/assets/img/products/header/460x215/${product.product.image_cover}`
        },
        {
          hid: `og:description`,
          property: 'og:description',
          content: product.short_description
        }
      ]
    }
  },
```

**Comment by NicoPennec** [comment]
It seems not a meta tags error. You have a more global issue on your server! Check your server logs.

I just run a `wget` command to quick check:

```sh
wget https://store.gameodds.gg/game/fantasy-bundle
--2017-11-06 21:51:56--  https://store.gameodds.gg/game/fantasy-bundle
Resolving store.gameodds.gg (store.gameodds.gg)... 104.25.53.102, 104.25.54.102, 2400:cb00:2048:1::6819:3566, ...
Connecting to store.gameodds.gg (store.gameodds.gg)|104.25.53.102|:443... connected.
HTTP request sent, awaiting response... 500 TypeError
2017-11-06 21:51:56 ERROR 500: TypeError.
```

Then you have a mixed content, your page includes HTTP resources from 'http://cdn.edgecast.steamstatic.com/

**Comment by atmar** [comment]
Hmm it's strange. It's not related to mixed content as a different product without that has the same error. I can't find any error so far either (using Engintron, nginx)

**Comment by NicoPennec** [comment]
I have a hypothesis for your error 500 (but I can't check your nginx error logs `/var/log/nginx/error.log`)

With wget or curl or from facebook test page, you request with the protocole http/1.
But from a browser (chrome, firefox, edge), you request with the protocole http/2.
So, maybe your server is not configured to fallbakc on http1 properly ???

Can you share an extract of your nginx config (from `/etc/nginx/sites-enabled/xxx` ) ?
and your `nuxt.config.js` file too?

**Comment by atmar** [comment]
There is no /etc/nginx/sites-enabled/xxx in this server. Engintron doesn't have it. Rules are set in /etc/nginx/custom_rules

https://github.com/engintron/engintron/wiki/Engintron-&-CloudFlare

```
if ($host ~ "store.gameodds.gg") {
 set $PROXY_DOMAIN_OR_IP "xxx.xx.xx.x";
 set $PROXY_TO_PORT 3000;
}
```
Traffic always goes through https with cloudflare option. Nginx error log is completely empty. I've reset it before trying the requests.

**Comment by atmar** [comment]
Found the issue. I couldn't run npm run dev on the production server for some reason so I only had a 500 error to work with. But I found a npm package called localtunnel https://localtunnel.github.io/www/ . It allowed me to expose my localhost application to the public. I've used the OG tester of Facebook on it and found the following error:
```
Cannot use function split on undefined
```

It showed me where it was located and I found the following code in my i18n plugin:

```
 let locale = req.headers['accept-language'].split(',')[0].split('-')[0].toLocaleLowerCase() || 'en'
```
Appearently there is no req.headers['accept-language'] when Facebook checks for the OG tags and it caused an error. Changed it to the following and everything works:
```
      let locale = 'en'
      if (req.headers['accept-language'] !== undefined) {
      let locale = req.headers['accept-language'].split(',')[0].split('-')[0].toLocaleLowerCase() || 'en'
      }
```
